sentry/src/main/java/io/sentry/util/StringUtils.java

```diff
@@ -72,4 +80,50 @@ private StringUtils() {}
     }
     return String.format(Locale.ROOT, "%.1f %cB", bytes / 1000.0, ci.current());
   }
+
+  /**
+   * Calculates the SHA-1 String hash
+   *
+   * @param str the String
+   * @param logger the Logger
+   * @return The hashed String or null otherwise
+   */
+  public static @Nullable String calculateStringHash(
+      final @Nullable String str, final @NotNull ILogger logger) {
+    if (str == null || str.isEmpty()) {
+      return null;
+    }
+
+    try {
+      // getInstance() method is called with algorithm SHA-1
+      final MessageDigest md = MessageDigest.getInstance("SHA-1");
+
+      // digest() method is called
+      // to calculate message digest of the input string
+      // returned as array of byte
+      final byte[] messageDigest = md.digest(str.getBytes(UTF_8));
+
+      // Convert byte array into signum representation
+      final BigInteger no = new BigInteger(1, messageDigest);
+
+      // Convert message digest into hex value
+      final StringBuilder stringBuilder = new StringBuilder(no.toString(16));
+
+      // Add preceding 0s to make it 32 bit
+      while (stringBuilder.length() < 32) {
+        stringBuilder.insert(0, "0");
+      }
+
+      // return the HashText
+      return stringBuilder.toString();
+    }
+
+    // For specifying wrong message digest algorithms
```

Comment:
this is here because of checked exceptions or do we know there are some Androids out there without SHA1?

When can SHA1 be "wrong"?

Reply:
Already answered on slack.